tools/toolchains/gcc.py

```diff
@@ -177,7 +192,13 @@ def get_compile_options(self, defines, includes, for_asm=False):
     @hook_tool
     def assemble(self, source, object, includes):
         # Build assemble command
-        cmd = self.asm + self.get_compile_options(self.get_symbols(True), includes) + ["-o", object, source]
+        cmd = self.asm
+
+        # Add coverage flag on files on which coverage is enabled.
+        if self.check_if_coverage_enabled(source):
```

Comment:
Coverage doesn't work on assembler code, see for example https://stackoverflow.com/questions/1367456/does-gcov-give-code-coverage-analysis-for-assembly-language-code